docs/codeql/ql-language-reference/ql-language-specification.rst

```diff
@@ -1948,38 +1949,42 @@ The store is first initialized with the *database content* of all built-in predi
 
 Each layer of the stratification is *populated* in order. To populate a layer, each predicate in the layer is repeatedly populated until the store stops changing. The way that a predicate is populated is as follows:
 
--  To populate a predicate that has a formula as a body, find all named tuples with identify each named tuple ``t`` that has the following properties:
-   - The tuple matches the body formula.
-   - The variables should be  the predicate's arguments.  
-   - If the predicate has a result, then the tuples should additionally have a value for ``result``
-   - If the predicate is a member predicate or characteristic predicate of a class ``C`` then the tuples should additionally have a value for ``this`` and each visible field on the class.
-   - The values corresponding to the arguments should all be a member of the declared types of the arguments.
-   - The values corresponding to ``result`` should all be a member of the result type.
-   - The values corresponding to the fields should all be a member of the declared types of the fields.
-   - If the predicate is a member predicate of a class ``C`` and not a characteristic predicate, then the tuples should additionally extend some tuple in ``C.class``. 
-   - If the predicate is a characteristic predicate of a class ``C``, then there should be a tuple ``t'``in ``C.extends`` such that for each visible field in ``C`` any field that is equal to or overrides a field in that ``t'`` should have the same value in ``t``. ``this`` should also map to the same value in ``t`` and ``t``.
+-  To populate a predicate that has a formula as a body, find each named tuple ``t`` that has the following properties:
```

Comment:
The initial sentence had some spurious words: I wasn't 100% sure which ones were right, but this is my best guess 😉